lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -0,0 +1,636 @@
+    aff4["query"] += ", ".join(
+        ["(unhex(md5(%s)), unhex(md5(%s)), %s, %s, %s)"] * (
+            len(aff4["args"]) / 5))
+
+    return [subjects, attributes, aff4]
+
+  def _ExecuteInsert(self, transaction):
+    """Get connection from pool and execute query"""
+    for query in transaction:
+      with self.pool.GetConnection() as cursor:
+        cursor.Execute(query["query"], query["args"])
+
+  def _ResolveRegex(self, subject, predicate_regex, token=None, timestamp=None, limit=None):
+    if limit and limit == 0:
+      return []
+
+    if isinstance(predicate_regex, str):
+      predicate_regex = [predicate_regex]
+
+    results = []
+    seen = set()
+
+    for regex in predicate_regex:
+      query, args = self._BuildQuery("SELECT", subject, regex, timestamp, limit, is_regex=True)
+      rows = self._ExecuteQuery(query, args)
+
+      for row in rows:
+        attribute = row["attribute"]
+        value = self._DecodeValue(attribute, row["value"])
+        results.append((attribute, value, row["timestamp"]))
+        if limit:
+          limit -= 1
+          if limit == 0:
+            return results
+
+    return results
+
+  def _CalculateAttributeStorageTypes(self):
+    """Build a mapping between column names and types."""
+    self.attribute_types = {}
+
+    for attribute in aff4.Attribute.PREDICATES.values():
+      self.attribute_types[attribute.predicate] = (
+          attribute.attribute_type.data_store_type)
+
+  def _EncodeValue(self, value):
+    """Encode the value for the attribute."""
+    if hasattr(value, "SerializeToString"):
+      return buffer(value.SerializeToString())
+    else:
+      # Types "string" and "bytes" are stored as strings here.
+      return buffer(utils.SmartStr(value))
+
+  def _DecodeValue(self, attribute, value):
+    required_type = self.attribute_types.get(attribute, "bytes")
+    if isinstance(value, buffer):
+      value = str(value)
+    if required_type in ("integer", "unsigned_integer"):
+      return int(value)
+    elif required_type == "string":
+      return utils.SmartUnicode(value)
+    else:
+      return value
+
+  def _BuildQuery(self, action, subject, predicate=None, timestamp=None,
+                  limit=None, is_regex=False):
+    """Build the SELECT query to be executed"""
+    args = []
+
+    fields = ""
+    criteria = "WHERE"
+    sorting = ""
+    tables = "FROM aff4"
+
+    subject = utils.SmartUnicode(subject)
+
+    #Set fields, tables, and criteria and append args
+    if is_regex:
+      tables += " JOIN attributes ON aff4.attribute_hash=attributes.hash"
+      regex = re.match(r'(^[a-zA-Z0-9_]+:[a-zA-Z0-9_:]*)(.*)', predicate)
+      if regex:
+        #If predicate has prefix then break down for query optimizations
+        parts = predicate.split(":", 1)
+
+        #Add prefix usage for all regex queries
+        criteria += " aff4.subject_hash=unhex(md5(%s)) AND aff4.prefix=(%s)"
```

Comment:
Is the "aff4.subject_hash=unhex(md5(%s))" piece what you are referring to?  That shouldn't be a problem to move.